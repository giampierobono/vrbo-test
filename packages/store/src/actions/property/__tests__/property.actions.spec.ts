import {
  failPollingPropertiesListAction,
  loadPropertiesAction,
  updatePropertiesListAction
} from "../property.actions";

describe("Property actions", () => {
  it("should return correct loadPropertiesAction object", () => {
    expect(loadPropertiesAction()).toMatchSnapshot();
  });

  it("should return correct updatePropertiesListAction object", () => {
    expect(
      updatePropertiesListAction([
        {
          rating: 10,
          ratingCount: 30,
          bath: 3,
          beds: 2,
          price: 100,
          title: "test prop",
          sqFeet: 200
        }
      ])
    ).toMatchSnapshot();
  });

  it("should return correct failPollingPropertiesListAction object", () => {
    expect(
      failPollingPropertiesListAction({ error: "fake error" })
    ).toMatchSnapshot();
  });
});
