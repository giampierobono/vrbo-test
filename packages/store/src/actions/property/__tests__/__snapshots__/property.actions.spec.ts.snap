// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property actions should return correct failPollingPropertiesListAction object 1`] = `
Object {
  "payload": Object {
    "error": "fake error",
  },
  "type": "[Property] Failure polling properties list",
}
`;

exports[`Property actions should return correct loadPropertiesAction object 1`] = `
Object {
  "type": "[Property] Poll properties list",
}
`;

exports[`Property actions should return correct updatePropertiesListAction object 1`] = `
Object {
  "payload": Array [
    Object {
      "bath": 3,
      "beds": 2,
      "price": 100,
      "rating": 10,
      "ratingCount": 30,
      "sqFeet": 200,
      "title": "test prop",
    },
  ],
  "type": "[Property] Update properties list",
}
`;
